f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml

```diff
@@ -39,106 +39,426 @@ Igce.prototype = {
+                item.classification_level = classLevel.toUpperCase();
+                // IGCE appends a CLASSFIED to the end of SECRET and TOP SECRET, format string to reflect that difference
+                if (item.classification_level == "SECRET" || item.classification_level == "TOP SECRET") {
+                    item.classification_level += " CLASSIFIED";
+                }
+
+            }
+        });
+
+        var formatted_clin_info = [];
+        clin_info.forEach(function (item1) {
+            clin_info.forEach(function (item2) {
+                var idiqClinString;
+                // If the items descriptions and classification_levels match, create a new idiqClinString
+                if (item1.description == item2.description && item1.classification_level == item2.classification_level && item1.idiq_clin != item2.idiq_clin) {
+                    // Lowest IDIQ CLIN first
+                    if (parseInt(item1.idiq_clin) < parseInt(item2.idiq_clin)) {
+                        idiqClinString = "x" + item1.idiq_clin.substr(1) + "/x" + item2.idiq_clin.substr(1) + " " + item1.description + " " + item1.classification_level;
+                    }
+                    else {
+                        idiqClinString = "x" + item2.idiq_clin.substr(1) + "/x" + item1.idiq_clin.substr(1) + " " + item1.description + " " + item1.classification_level;
+                    }
+                    // If it doesn't exist in the array of new CLINs, push it
+                    if (formatted_clin_info.indexOf(idiqClinString) == -1) {
+                        formatted_clin_info.push(idiqClinString);
+                    }
+                }
+                // If the description is Travel or External Agency Fee, this is a one-off IDIQ CLIN
+                else if (item1.description == "Travel" || item1.description == "External Ordering Agency Fee") {
+                    idiqClinString = "x" + item1.idiq_clin.substr(1) + " " + item1.description + " " + item1.contract_type;
+                    if (formatted_clin_info.indexOf(idiqClinString) == -1) {
+                        formatted_clin_info.push(idiqClinString);
+                    }
+                }
             });
         });
+        return formatted_clin_info;
+    },
+    getPayload: function () {
+        var acq_package = this.acquisitionPackageSysId;
+        var periodOfPerformance = acq_package.period_of_performance;
+        var currentEnvironment = acq_package.current_environment;
+        var selectedServiceOfferings = acq_package.selected_service_offerings;
+        var idiqClins = acq_package.idiq_clins;
+        var fundingRequest = acq_package.funding_request;
+        var requirementsCostEstimate = acq_package.requirements_cost_estimate;
 
 
-        return this.periods_estimate;
+        /////////////////////////////////////////////////////////////////////////////////////////////////////////////
+        // TODO: 
```

Comment:
I like the sectioning off of the `TODO:`,  is the thought to add more here in the future?

Reply:
It seems like the intent of the TODO is to provide the null checks deeper in the data structure.  For now they're left at a high level.

Reply:
Yeah I was hoping we could get a unified error response in the future when we rework the IGCE.

There is the `ErrorHandler` script include, however I went way out of scope for this ticket and spent too much time refactoring to spend any extra time setting that up with the `Igce` script.